csharp_dotnetcore/Dialogs/FoodBotDialogSet.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.Dialogs.Choices;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace FoodBot.Dialogs
+{
+    public class FoodBotDialogSet : DialogSet
+    {
+        const string MENUPROMPT = "menuPrompt";
+        const string DONATEFOODDIALOG = "donateFoodDialog";
+        const string FINDFOODDIALOG = "findFoodDialog";
+        const string CONTACTDIALOG = "contactDialog";
+
+        public static string MenuDialogId => "menuDialog";
+
+        public FoodBotDialogSet(IStatePropertyAccessor<DialogState> dialogStatePropertyAccessor)
+            : base(dialogStatePropertyAccessor)
+        {
+            // Adds a waterfall dialog that prompts users for the top level menu to the dialog set
+            // Define the steps of the waterfall dialog and add it to the set.
+            Add(new WaterfallDialog(
```

Comment:
This is better because it's moved out of the bot, but now it's technically part of the custom `DialogSet` which I still find as breaking SRP. I would personally go one step further and move this to its own `ComponentDialog` class. This would also align more closely with what I think @ryanvolum was going to do on the node side after feedback session yesterday.